source/postprocess/boundary_velocity_residual_statistics.cc

```diff
@@ -0,0 +1,419 @@
+                                                                                         .translate_id_to_symbol_name (p->first))
+                                           + " (m/s)";
+              statistics.add_value (name_max, p->second);
+              const std::string name_min = "Minimum residual velocity magnitude on boundary with indicator "
+                                           + Utilities::int_to_string(a->first)
+                                           + aspect::Utilities::parenthesize_if_nonempty(this->get_geometry_model()
+                                                                                         .translate_id_to_symbol_name (a->first))
+                                           + " (m/s)";
+              statistics.add_value (name_min, a->second);
+              const std::string name_rms = "RMS residual velocity on boundary with indicator "
+                                           + Utilities::int_to_string(rms->first)
+                                           + aspect::Utilities::parenthesize_if_nonempty(this->get_geometry_model()
+                                                                                         .translate_id_to_symbol_name (rms->first))
+                                           + " (m/s)";
+              statistics.add_value (name_rms, rms->second);
+              // also make sure that the other columns filled by the this object
+              // all show up with sufficient accuracy and in scientific notation
+              statistics.set_precision (name_max, 8);
+              statistics.set_scientific (name_max, true);
+              statistics.set_precision (name_min, 8);
+              statistics.set_scientific (name_min, true);
+              statistics.set_precision (name_rms, 8);
+              statistics.set_scientific (name_rms, true);
+            }
+
+          // finally have something for the screen
+          screen_text.precision(4);
+          if (this->convert_output_to_years() == true)
+            {
+              screen_text << p->second *year_in_seconds << " m/yr, "
+                          << a->second *year_in_seconds << " m/yr, "
+                          << rms->second *year_in_seconds << " m/yr"
+                          << (index == global_max_vel.size()-1 ? "" : ", ");
+            }
+          else
+            {
+              screen_text << p->second << " m/s, "
+                          << a->second << " m/s, "
+                          << rms->second << " m/s"
+                          << (index == global_max_vel.size()-1 ? "" : ", ");
+            }
+
+        }
+
+      return std::pair<std::string, std::string> ("Max, min, and rms residual velocity along boundary parts:",
+                                                  screen_text.str());
+    }
+
+
+
+    template <int dim>
+    void
+    BoundaryVelocityResidualStatistics<dim>::initialize ()
```

Comment:
Please move this function to the beginning of the file. Our usual convention (not strictly followed, but a good convention) is the following order of functions in a plugin:

1. constructor (if any)
2. destructor (if any)
3. initialize (if present)
4. any plugin specific function (`evaluate` in this case)
5. the main function of the plugin (`execute` in this case)
6. declare_parameters (if present)
7. parse_parameters (if present)